sdk/client/task_resource.go

```diff
@@ -29,6 +30,8 @@ type TaskResourceApiService struct {
 	*APIClient
 }
 
+var hostname, _ = os.Hostname()
```

Comment:
**suggestion:** Right now, this runs a system call on startup in any binary where this package is imported, whether any of the functionality in the package is used or not. Running a system call as the side-effect of an import is rather unexpected, and it's not something that the user can disable.

I think that it would make more sense to fetch this value lazily only the first time it's needed. [sync.Once](https://pkg.go.dev/sync#Once) can be used for this purpose.

Reply:
Got it! Should be solved by next commit :)

Reply:
Once it's solved this LGTM. 👍